### PR TITLE
Move the workload e2e tests to it's own package

### PR DESCRIPTION
**Pull request by wanghaoran1988**
**What this PR does / why we need it**:

Move the workload e2e tests to it's own package

**Release note**:
```
None
```

### CODE REVIEW

**Comment by kargakis** [comment]
@kubernetes/sig-apps-pr-reviews

**Comment by kargakis** [comment]
@k8s-bot ok to test

**Comment by kargakis** [comment]
Part of https://github.com/kubernetes/kubernetes/issues/7666

**Comment by kargakis** [comment]
Alongside with this move we need to create a new workload-specific package where all the utilities will live. Currently we have utilities in two different places:
https://github.com/kubernetes/kubernetes/tree/master/test/utils
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go
I'd vote for a `test/utils/workloads` package.

**Comment by wanghaoran1988** [comment]
@kargakis Good points, will update tomorrow :)

**Comment by wanghaoran1988** [comment]
@k8s-bot pull-kubernetes-bazel test this

**Comment by wanghaoran1988** [comment]
@kargakis @ixdy Would you mind tell me why the bazel build failed, and I already run hack/update-bazel.sh .

**Comment by wanghaoran1988** [comment]
> I'd vote for a test/utils/workloads package.

@kargakis I'd prefer move them to test/e2e/framework, so that other components can also use them conveniently like how we do currently, let me know you have any other concerns.

**Comment by kargakis** [comment]
framework is already bloated; workload-specific utilities should be in their own package. OWNERS files are not valuable when the bot has to pick somebody out of the whole organization.

**Comment by fabianofranz** [comment]
@k8s-bot pull-kubernetes-unit test this
@k8s-bot pull-kubernetes-bazel test this

**Comment by wanghaoran1988** [comment]
@kargakis I am trying to work as you suggested, but it's hard to split the workloads func from the framework/util.go, there are some basic/common func in framework package that workload related funcs will use, after we split the workload related func to test/utils/workloads, this new package have to depend on the framework package, and some other resource utils under framework package need use the workloads func, this will introduce the cycle import in golang, it's really difficult to split clearly now, so to keep this pr simple, how about  address two problems in this pr:
1. E2e tests move to it's own package
2. Split every workload related resource utils have it's own util go file, deployment.go rc.go statefu_set.go .etc. this will make it a little easier to maintain currently.

**Comment by kargakis** [comment]
@wanghaoran1988 I am ok with moving just the tests if we need to do more work in order to split the workload-specific utilities.

**Comment by wanghaoran1988** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by wanghaoran1988** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by kargakis** [comment]
@wanghaoran1988 that job is broken and does not affect the queue I think.

**Comment by wanghaoran1988** [comment]
@kargakis Thanks :)

File: test/e2e/workload/OWNERS [hunk 1]
```diff
@@ -0,0 +1,10 @@
+approvers:
+- janetkuo
+- nikhiljindal
+- kargakis
+- mfojtik
+reviewers:
+- janetkuo
```
Comment (kargakis) [inline test/e2e/workload/OWNERS]: Add sig-apps-reviewers

Comment (wanghaoran1988) [inline test/e2e/workload/OWNERS]: @kargakis Should delete the other people, and only "sig-apps-reviewers" is enough ?

Comment (kargakis) [inline test/e2e/workload/OWNERS]: Yes

Comment (wanghaoran1988) [inline test/e2e/workload/OWNERS]: @kargakis Updated

File: test/e2e/kubectl.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 	"k8s.io/kubernetes/test/e2e/generated"
 	"k8s.io/kubernetes/test/e2e/scheduling"
+	. "k8s.io/kubernetes/test/e2e/workload"
```
Comment (kargakis) [inline test/e2e/kubectl.go]: Why do we need this import here?

Comment (kargakis) [inline test/e2e/kubectl.go]: At the very lease, don't import with . so we can know where is the package used.

Comment (wanghaoran1988) [inline test/e2e/kubectl.go]: @kargakis Done

File: test/e2e/framework/rs_util.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	By(fmt.Sprintf("deleting ReplicaSet %s in namespace %s", name, ns))
+	rc, err := clientset.Extensions().ReplicaSets(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			Logf("ReplicaSet %s was already deleted: %v", name, err)
+			return nil
+		}
+		return err
+	}
+	reaper, err := kubectl.ReaperFor(extensionsinternal.Kind("ReplicaSet"), internalClientset)
```
Comment (kargakis) [inline test/e2e/framework/rs_util.go]: Open a separate issue to remove the cli reaper from here and instead do a server-side deletion

File: test/e2e/kubectl.go [hunk 4]
```diff
@@ -205,7 +207,7 @@ var _ = framework.KubeDescribe("Kubectl alpha client", func() {
 		})
 
 		It("should create a ScheduledJob", func() {
-			framework.SkipIfMissingResource(f.ClientPool, ScheduledJobGroupVersionResource, f.Namespace.Name)
+			framework.SkipIfMissingResource(f.ClientPool, workload.ScheduledJobGroupVersionResource, f.Namespace.Name)
```
Comment (kargakis) [inline test/e2e/kubectl.go]: Instead of importing the whole package just for these two variables, copy them here.

Comment (wanghaoran1988) [inline test/e2e/kubectl.go]: hmm, sounds reasonable.

File: test/e2e/framework/rs_util.go (hunk 3 above)
Comment (wanghaoran1988) [inline test/e2e/framework/rs_util.go]: opened here:https://github.com/kubernetes/kubernetes/issues/47832

**Comment by wanghaoran1988** [comment]
@kargakis Hi, all comments are addressed, would you mind add lgtm label, as every time I do a rebase, it will cost a lot time.

**Comment by kargakis** [comment]
/lgtm

Note that this won't get merged before the code freeze is lifted

**Comment by kargakis** [comment]
@wanghaoran1988 code freeze is lifted, we can merge this as soon as you rebase one more time :)

**Comment by wanghaoran1988** [comment]
@kargakis Cool, updated

**Comment by wanghaoran1988** [comment]
@kargakis Hoo, all tests passed now

**Comment by kargakis** [comment]
Hooray! Thanks a lot @wanghaoran1988 !

**Pull request merged**